### PR TITLE
BUG-108475 Image Catalog v3

### PR DESCRIPTION
**Pull request by topolyai5**

### CODE REVIEW

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v3/ImageCatalogV3Endpoint.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = ImageCatalogOpDescription.CREATE_IN_ORG, produces = ContentType.JSON, notes = Notes.IMAGE_CATALOG_NOTES,
+            nickname = "createClusterInOrganization")
+    ImageCatalogResponse createInOrganization(@PathParam("organizationId") Long organizationId, @Valid ImageCatalogRequest request);
+
+    @DELETE
+    @Path("{name}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = ImageCatalogOpDescription.DELETE_BY_NAME_IN_ORG, produces = ContentType.JSON, notes = Notes.IMAGE_CATALOG_NOTES,
+            nickname = "deleteClusterInOrganization")
```
Comment (mhmxs) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v3/ImageCatalogV3Endpoint.java]: nicknames are wrong

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/EndpointConfig.java [hunk 2]
```diff
@@ -108,6 +108,19 @@ private void registerEndpoints() {
         register(SubscriptionController.class);
         register(UserController.class);
         register(UtilController.class);
+        register(RdsConfigController.class);
```
Comment (mhmxs) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/EndpointConfig.java]: Some records are duplicated

File: core-api/src/main/java/com/sequenceiq/cloudbreak/client/CloudbreakClient.java [hunk 3]
```diff
@@ -189,6 +190,10 @@ public UtilEndpoint utilEndpoint() {
         return refreshIfNeededAndGet(UtilEndpoint.class);
     }
 
+    public ImageCatalogV3Endpoint imageCatalogV3Endpoint() {
```
Comment (attilapalfi92) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/client/CloudbreakClient.java]: I know it's not strictly part of your task but can you please add the other already implemented V3 endpoints here?

Comment (mhmxs) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/client/CloudbreakClient.java]: At the end we have to check both CloudbreakClient and newway client to, so at this point it is not necessary i think

Comment (attilapalfi92) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/client/CloudbreakClient.java]: yep thats also good

File: core-api/src/main/java/com/sequenceiq/cloudbreak/doc/ControllerDescription.java [hunk 4]
```diff
@@ -41,6 +41,7 @@
     public static final String PROXY_CONFIG_V3_DESCRIPTION = "Operations on proxy config by organization";
     public static final String STACK_V3_DESCRIPTION = "Operations on stack by organization";
     public static final String MANAGEMENT_PACK_V3_DESCRIPTION = "Operations on Management pack by Organization";
+    public static final String IMAGE_CATALOG_V3_DESCRIPTION = "Operations on Cluster by Organization";
```
Comment (mhmxs) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/doc/ControllerDescription.java]: Text contains `Cluster`

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/ImageCatalogV3Controller.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        return null;
+    }
+
+    @Override
+    public ImageCatalogResponse getByNameInOrganization(Long organizationId, String name) {
+        return null;
+    }
+
+    @Override
+    public ImageCatalogResponse createInOrganization(Long organizationId, @Valid ImageCatalogRequest request) {
```
Comment (mhmxs) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/ImageCatalogV3Controller.java]: @Valid not necessary here as i know

**Pull request merged**